File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> found = new HashMap<>();
        int[] ans = new int[2];
        for(int i=0; i<nums.length; i++){
            if(null == found.get(nums[i])){
                found.put(target - nums[i], i);
            }else{
                ans[0] = found.get(nums[i]);
                ans[1] = i;
            }
        }
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        s = set()
        d = {}
        for i, n in enumerate(nums):
            if n in s:
                return [d[n], i]
            diff = target - n
            s.add(diff)
            d[diff] = i
            